src/Parser.php

```diff
@@ -104,22 +112,31 @@ public function parse($input, $file = null)
      *
      * @return array
      *
-     * @phpstan-return TToken
+     * @phpstan-return (
+     *     $type is 'TableRow'
+     *         ? TTableRowToken
+     *         : ($type is 'Tag'
+     *             ? TTagToken
+     *             : ($type is 'Step'
+     *                 ? TStepToken
+     *                 : ($type is TGeneralKeywordsType
+     *                     ? TKeywordToken
+     *                     : TToken)))
+     * )
      *
      * @throws ParserException
      */
     protected function expectTokenType($type)
     {
-        $types = (array) $type;
```

Comment:
I think this change should be safebecause:
1. phpdoc param was always string and it never said anything about possibly passing an array
2. currently we never pass an array - only a single string